src/index.js

```diff
@@ -12,35 +12,27 @@ export { Bibliography };
 
 const __dirname = fileURLToPath(new URL("..", import.meta.url));
 
-function defaultRenderCitation (citationTemplate) {
-	if (citationTemplate) {
-		return info => nunjucks.render(citationTemplate, info);
-	}
-	else {
-		let template = fs.readFileSync(__dirname + "/_includes/_citations.njk", "utf8");
-		return info => nunjucks.renderString(template, info);
-	}
-}
-
 let doiTemplates = {
 	url: '<a href="$&" class="doi">$&</a>',
 	id: '<a href="https://doi.org/$1" class="doi">$1</a>',
 };
 
 export default function (config, {
-	citationTemplate,
-	citationRender = defaultRenderCitation(citationTemplate),
+	citationTemplate = __dirname + "/_includes/_citations.njk",
+	citationRender,
 	style, locale, // defaults set in Bibliography
 	bibliography: globalBibliography,
 } = {}) {
 	const references = new Bibliographies({globalBibliography, style, locale});
+	let fileRenderer;
+	const render = citationRender ?? (async function(...args){
+		fileRenderer ??= await RenderPlugin.File(citationTemplate);
```

Comment:
I suspect there’s some magic at play but if we’re not passing the 11ty config to this somehow, how does it "know" what templating settings this particular instance of 11ty is using?

Reply:
Good point - it _looks_ like it re-uses the same config. I briefly tested this from creating a custom filter in the demo's `.eleventy.js` and it is indeed useable in the `_references.njk` file. I tried looking at Eleventy's source code, but I didn't learn a whole lot, so I'll just say it's magic 😄